securitypolicy/securitypolicy.go

```diff
@@ -23,15 +23,16 @@ import (
 // The zero value of this struct will use SecurityPolicy#None although
 // using in this manner is discouraged for readability
 type EncryptionAlgorithm struct {
-	blockSize       int
-	minPadding      int
-	encrypt         func(cleartext []byte) (ciphertext []byte, err error)
-	decrypt         func(ciphertext []byte) (cleartext []byte, err error)
-	signature       func(message []byte) (signature []byte, err error)
-	verifySignature func(message, signature []byte) error
-	signatureLength int
-	encryptionURI   string
-	signatureURI    string
+	blockSize           int
```

Comment:
Can you refactor this as a struct with methods instead of function variables?

Reply:
I'll try refactor this into interfaces and see how it looks.  I originally liked the closure concept because it allowed the encryption to be initialized without keeping the keys stored in config objects, but I admit it's not as clear as it could be.